### PR TITLE
Use a page reader for every "show" commands on dell powerconnect

### LINKED ISSUE #178: Netman should support pagination
We saw Netman wait forever when we display an interface configuration and the uplink port channel is huge.

# Command
> show running-config interface port-channel 10

# CoS
* Netman doesn't wait forever when the switch asks to hit a key to continue the pagination

### PR DESCRIPTION
**Pull request by lindycoder**
Any show command can have pagination so now they will all be run
in a page reader that abstracts the pagination

Fixes #178

### CODE REVIEW

**Comment by fbouliane** [comment]
:+1: nice job to refactor the pagereader.

File: tests/adapters/switches/page_reader_test.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        self.shell_mock.should_receive("do").with_args("command", wait_for=("--More-- or (q)uit", "#"), include_last_line=True).once().ordered().and_return([
+            "line1",
+            "line2",
+            "--More-- or (q)uit"
+        ])
+
+        self.shell_mock.should_receive("send_key").with_args("m", wait_for=("--More-- or (q)uit", "#"), include_last_line=True).once().ordered().and_return([
+            "line3",
+            "line4",
+            "--More-- or (q)uit"
```
Comment (mat128) [inline tests/adapters/switches/page_reader_test.py]: Isnt there one more blank line before/after this prompt?

**Comment by mat128** [comment]
inline comment + failing tests, but good job on the PageReader

File: tests/adapters/switches/page_reader_test.py (hunk 1 above)
Comment (lindycoder) [inline tests/adapters/switches/page_reader_test.py]: the shell client never returns blank lines

**Comment by lindycoder** [comment]
comments answered and texts fixed

**Pull request merged**